pysnark/fixedpoint.py

```diff
@@ -28,107 +28,171 @@
 # ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 # POSSIBILITY OF SUCH DAMAGE.
 
+import warnings
 import pysnark.runtime
-from pysnark.runtime import LinComb, is_base_value, PrivVal, PubVal
+from pysnark.runtime import LinComb, PrivVal, PubVal, ConstVal
 
 """
 Support for fixed-point computations
+Costs 0 constraints to convert a LinComb to a LinCombFxp
+A float x is represented as the integer x * (1 << res)
 """
 
-class LinCombFxp:
-    """
-    Variable representing a fixed-point number. Number x is represented as integer
-    :math:`x * 2^r`, where r is the resolution :py:data:`VarFxp.res`
-    """
-    
-    res = 8
+# Fixed point resolution
+resolution = 8
 
+class LinCombFxp:
     def __init__(self, lc):
-        if not isinstance(lc, LinComb): raise RuntimeError("wrong type for LinCombFxp")
+        if not isinstance(lc, LinComb):
+            raise RuntimeError("Wrong type for LinCombFxp")
         self.lc = lc
-
-    @classmethod
-    def fromvar(cls, var, doconvert):
-        if not isinstance(var, LinComb): raise RuntimeError("wrong type for fromvar")
-        if doconvert:
-            return cls(var*(1<<LinCombFxp.res))
-        else:
-            return cls(var)
     
     @classmethod
-    def tofloat(self, val):
-        return float(val)/(1<<LinCombFxp.res)
+    def add_scaling(cls, val):

```

Comment:
I like adding this function, it definitely makes it more easy to understand what is going on as opposed to the previous conversion routines...